### ISSUE TITLE
Trailing comma on the package.json crash server

### ISSUE THREAD

**Issue by GramThanos** [typegoose/mongodb-memory-server#906]
## What is the Problem?

The server does not crash if there is a trailing comma on the package.json. Although this is not a valid JSON, ~since NPM accepts it~ (edit: my bad it does not), maybe mongodb-memory-server should consider accepting trailing commas.

## Code Example

```json
{
  "name": "test",
  "version": "1.0.0",
  "main": "start.ts",
  "type": "module",
  "scripts": {
    "start": "bun start.ts"
  },
  "author": "",
  "license": "ISC",
  "description": "",
  "dependencies": {
    "mongodb-memory-server": "^10.1.2"
  },
  "config": {
    "mongodbMemoryServer": {
      "debug": "1",
    }
  }
}

```

## Do you know *why* it happenes?

The function [findPackageJson](https://github.com/typegoose/mongodb-memory-server/blob/0b65dd117def84804da63281544a59afca0d5056/packages/mongodb-memory-server-core/src/util/resolveConfig.ts#L84) fails to parse the contents of package.json if there is a trailing comma inside.

**Comment by hasezoey** [comment]
`findPackageJson` is literally just using default JS `JSON.parse`:
https://github.com/typegoose/mongodb-memory-server/blob/0b65dd117def84804da63281544a59afca0d5056/packages/mongodb-memory-server-core/src/util/resolveConfig.ts#L84

And as the js `JSON.parse` object/function does not provide configurability (AFAIK) and [a trailing comma is not valid JSON](https://stackoverflow.com/questions/201782/can-you-use-a-trailing-comma-in-a-json-object), i will not change the parser for just this one parsing step.
Though i might just add a try-catch around that block to not crash but warn instead, would that be acceptable in this case?

TL;DR: i will not change the parser (to allow trailing commas), but might add a try-catch around it.

> since NPM accepts it

I cant find a reference for that from my quick search, is this something npm is supposed to support, or does npm just not complain currently?

**Comment by GramThanos** [comment]
@hasezoey my bad, I did some more tests and NPM also throws an error.
I am closing this, I am sorry for the spam.

Just, consider removing the comma from the enable debug mode example.

**Comment by hasezoey** [comment]
> Just, consider removing the comma from the enable debug mode example.

Thanks for pointing this out, will update.
